File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.js
let input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const n = input.shift()
const inputList = input


inputList.forEach(list => {
    let fncCount = 0;
    const recursion = (s, l,  r) => {
        fncCount += 1;
        if(l >= r) return 1;
        else if(s[l] != s[r]) return 0;
        else return recursion(s, l+1, r-1);
    }
    
    const isPalindrome = (s) => {
        return recursion(s, 0, s.length-1);
    }
    console.log(isPalindrome(list),fncCount)
})

File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.py
N = int(input())


def recursion(s, l, r, depth):
    if l >= r:
        return [1, depth]
    elif s[l] != s[r]:
        return [0, depth]
    else:
        return recursion(s, l + 1, r - 1, depth + 1)


def isPalindrome(s):
    return recursion(s, 0, len(s) - 1, 1)


for i in range(N):
    s = list(input())
    result, count = isPalindrome(s)
    print(result, count)
